The build failed

This test failed!

To configure my behavior, see [the Flaky Bot documentation](https://github.com/googleapis/repo-automation-bots/tree/main/packages/flakybot).

If I'm commenting on this issue too often, add the `flakybot: quiet` label and
I will stop commenting.

---

commit: aefd33b4e9d614247c516b43f9bb8f897231963f
buildURL: [Build Status](https://source.cloud.google.com/results/invocations/fc5cddc4-a6ac-46fc-a6c1-d0d85c8c3f1c), [Sponge](http://sponge2/fc5cddc4-a6ac-46fc-a6c1-d0d85c8c3f1c)
status: failed
<details><summary>Test output</summary><br><pre>java.lang.NullPointerException
	at gce.diregapic.samples@1.0-SNAPSHOT/compute.SnippetsIT.cleanup(SnippetsIT.java:189)
</pre></details>

```
#
# compute/cloud-client
#

[ERROR] Tests run: 2, Failures: 1, Errors: 1, Skipped: 0, Time elapsed: 983.978 s <<< FAILURE! - in compute.SnippetsIT
[ERROR] compute.SnippetsIT  Time elapsed: 983.969 s  <<< FAILURE!
com.google.common.truth.AssertionErrorWithFacts:
expected to contain:
    firewall-rule-118d845d-ac30-466a-9e03-576b9a005108
but was:
    default-ba5h4uquy4cktbsldj6ba2g3
    default-ba5h4uquy4cktbsldj6ba2g3-v6
    default-iyvrmjzsszuzmmbd5kdofnu3-1
    default-iyvrmjzsszuzmmbd5kdofnu3-2
    default-iyvrmjzsszuzmmbd5kdofnu3-3
    ....
```

--- a/compute/cloud-client/src/test/java/compute/SnippetsIT.java
+++ b/compute/cloud-client/src/test/java/compute/SnippetsIT.java
@@ -328,7 +328,7 @@ public class SnippetsIT {
     } catch (NotFoundException e) {
       System.out.println("Rule already deleted ! ");
       return true;
-    } catch (InvalidArgumentException e) {
+    } catch (InvalidArgumentException | NullPointerException e) {
       System.out.println("Rule is not ready (probably being deleted).");
       return true;
     }

--- a/compute/cloud-client/src/test/java/compute/SnippetsIT.java
+++ b/compute/cloud-client/src/test/java/compute/SnippetsIT.java
@@ -291,8 +291,8 @@ public class SnippetsIT {
 
   public static void testListFirewallRules()
       throws IOException, ExecutionException, InterruptedException {
-    PrintStream out = System.out;
     ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+    PrintStream out = System.out;
     System.setOut(new PrintStream(stdOut));
     if (!isFirewallRuleDeletedByGceEnforcer(PROJECT_ID, FIREWALL_RULE_CREATE)) {
       compute.ListFirewallRules.listFirewallRules(PROJECT_ID);
@@ -306,8 +306,8 @@ public class SnippetsIT {
 
   public static void testPatchFirewallRule()
       throws IOException, InterruptedException, ExecutionException {
-    PrintStream out = System.out;
     ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+    PrintStream out = System.out;
     System.setOut(new PrintStream(stdOut));
     if (!isFirewallRuleDeletedByGceEnforcer(PROJECT_ID, FIREWALL_RULE_CREATE)) {
       try (FirewallsClient client = FirewallsClient.create()) {

--- a/compute/cloud-client/src/test/java/compute/SnippetsIT.java
+++ b/compute/cloud-client/src/test/java/compute/SnippetsIT.java
@@ -291,30 +291,34 @@ public class SnippetsIT {
 
   public static void testListFirewallRules()
       throws IOException, ExecutionException, InterruptedException {
+    ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
     if (!isFirewallRuleDeletedByGceEnforcer(PROJECT_ID, FIREWALL_RULE_CREATE)) {
-      ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
-      System.setOut(new PrintStream(stdOut));
       compute.ListFirewallRules.listFirewallRules(PROJECT_ID);
       assertThat(stdOut.toString()).contains(FIREWALL_RULE_CREATE);
-      stdOut.close();
-      System.setOut(null);
     }
+    // Clear system output to not affect other tests.
+    // Refrain from setting out to null.
+    stdOut.close();
+    System.setOut(new PrintStream(new ByteArrayOutputStream()));
   }
 
   public static void testPatchFirewallRule()
       throws IOException, InterruptedException, ExecutionException {
+    ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+    System.setOut(new PrintStream(stdOut));
     if (!isFirewallRuleDeletedByGceEnforcer(PROJECT_ID, FIREWALL_RULE_CREATE)) {
       try (FirewallsClient client = FirewallsClient.create()) {
-        ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
-        System.setOut(new PrintStream(stdOut));
         Assert.assertEquals(1000, client.get(PROJECT_ID, FIREWALL_RULE_CREATE).getPriority());
         compute.PatchFirewallRule.patchFirewallPriority(PROJECT_ID, FIREWALL_RULE_CREATE, 500);
         TimeUnit.SECONDS.sleep(5);
         Assert.assertEquals(500, client.get(PROJECT_ID, FIREWALL_RULE_CREATE).getPriority());
-        stdOut.close();
-        System.setOut(null);
       }
     }
+    // Clear system output to not affect other tests.
+    // Refrain from setting out to null as it will throw NullPointer in the subsequent tests.
+    stdOut.close();
+    System.setOut(new PrintStream(new ByteArrayOutputStream()));
   }
 
   public static boolean isFirewallRuleDeletedByGceEnforcer(String projectId,

--- a/compute/cloud-client/src/test/java/compute/SnippetsIT.java
+++ b/compute/cloud-client/src/test/java/compute/SnippetsIT.java
@@ -291,6 +291,7 @@ public class SnippetsIT {
 
   public static void testListFirewallRules()
       throws IOException, ExecutionException, InterruptedException {
+    PrintStream out = System.out;
     ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
     System.setOut(new PrintStream(stdOut));
     if (!isFirewallRuleDeletedByGceEnforcer(PROJECT_ID, FIREWALL_RULE_CREATE)) {
@@ -300,11 +301,12 @@ public class SnippetsIT {
     // Clear system output to not affect other tests.
     // Refrain from setting out to null.
     stdOut.close();
-    System.setOut(new PrintStream(new ByteArrayOutputStream()));
+    System.setOut(out);
   }
 
   public static void testPatchFirewallRule()
       throws IOException, InterruptedException, ExecutionException {
+    PrintStream out = System.out;
     ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
     System.setOut(new PrintStream(stdOut));
     if (!isFirewallRuleDeletedByGceEnforcer(PROJECT_ID, FIREWALL_RULE_CREATE)) {
@@ -318,7 +320,7 @@ public class SnippetsIT {
     // Clear system output to not affect other tests.
     // Refrain from setting out to null as it will throw NullPointer in the subsequent tests.
     stdOut.close();
-    System.setOut(new PrintStream(new ByteArrayOutputStream()));
+    System.setOut(out);
   }
 
   public static boolean isFirewallRuleDeletedByGceEnforcer(String projectId,

--- a/compute/cloud-client/src/test/java/compute/SnippetsIT.java
+++ b/compute/cloud-client/src/test/java/compute/SnippetsIT.java
@@ -167,7 +167,9 @@ public class SnippetsIT {
     requireEnvVar("GOOGLE_APPLICATION_CREDENTIALS");
     requireEnvVar("GOOGLE_CLOUD_PROJECT");
 
-    deleteFirewallRuleIfNotDeletedByGceEnforcer(PROJECT_ID, FIREWALL_RULE_CREATE);
+    if (!isFirewallRuleDeletedByGceEnforcer(PROJECT_ID, FIREWALL_RULE_CREATE)) {
+      DeleteFirewallRule.deleteFirewallRule(PROJECT_ID, FIREWALL_RULE_CREATE);
+    }
     compute.DeleteInstance.deleteInstance(PROJECT_ID, ZONE, MACHINE_NAME_ENCRYPTED);
     compute.DeleteInstance.deleteInstance(PROJECT_ID, ZONE, MACHINE_NAME);
     compute.DeleteInstance.deleteInstance(PROJECT_ID, ZONE, MACHINE_NAME_LIST_INSTANCE);
@@ -287,30 +289,35 @@ public class SnippetsIT {
         .encodeToString(stringBuilder.toString().getBytes(StandardCharsets.US_ASCII));
   }
 
-  public static void testListFirewallRules() throws IOException {
-    ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
-    System.setOut(new PrintStream(stdOut));
-    compute.ListFirewallRules.listFirewallRules(PROJECT_ID);
-    assertThat(stdOut.toString()).contains(FIREWALL_RULE_CREATE);
-    stdOut.close();
-    System.setOut(null);
-  }
-
-  public static void testPatchFirewallRule()
-      throws IOException, InterruptedException, ExecutionException {
-    try (FirewallsClient client = FirewallsClient.create()) {
+  public static void testListFirewallRules()
+      throws IOException, ExecutionException, InterruptedException {
+    if (!isFirewallRuleDeletedByGceEnforcer(PROJECT_ID, FIREWALL_RULE_CREATE)) {
       ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
       System.setOut(new PrintStream(stdOut));
-      Assert.assertEquals(1000, client.get(PROJECT_ID, FIREWALL_RULE_CREATE).getPriority());
-      compute.PatchFirewallRule.patchFirewallPriority(PROJECT_ID, FIREWALL_RULE_CREATE, 500);
-      TimeUnit.SECONDS.sleep(5);
-      Assert.assertEquals(500, client.get(PROJECT_ID, FIREWALL_RULE_CREATE).getPriority());
+      compute.ListFirewallRules.listFirewallRules(PROJECT_ID);
+      assertThat(stdOut.toString()).contains(FIREWALL_RULE_CREATE);
       stdOut.close();
       System.setOut(null);
     }
   }
 
-  public static void deleteFirewallRuleIfNotDeletedByGceEnforcer(String projectId,
+  public static void testPatchFirewallRule()
+      throws IOException, InterruptedException, ExecutionException {
+    if (!isFirewallRuleDeletedByGceEnforcer(PROJECT_ID, FIREWALL_RULE_CREATE)) {
+      try (FirewallsClient client = FirewallsClient.create()) {
+        ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
+        System.setOut(new PrintStream(stdOut));
+        Assert.assertEquals(1000, client.get(PROJECT_ID, FIREWALL_RULE_CREATE).getPriority());
+        compute.PatchFirewallRule.patchFirewallPriority(PROJECT_ID, FIREWALL_RULE_CREATE, 500);
+        TimeUnit.SECONDS.sleep(5);
+        Assert.assertEquals(500, client.get(PROJECT_ID, FIREWALL_RULE_CREATE).getPriority());
+        stdOut.close();
+        System.setOut(null);
+      }
+    }
+  }
+
+  public static boolean isFirewallRuleDeletedByGceEnforcer(String projectId,
       String firewallRule) throws IOException, ExecutionException, InterruptedException {
     /* (**INTERNAL method**)
       This method will prevent test failure if the firewall rule was auto-deleted by GCE Enforcer.
@@ -318,12 +325,14 @@ public class SnippetsIT {
      */
     try {
       GetFirewallRule.getFirewallRule(projectId, firewallRule);
-      DeleteFirewallRule.deleteFirewallRule(projectId, firewallRule);
     } catch (NotFoundException e) {
       System.out.println("Rule already deleted ! ");
+      return true;
     } catch (InvalidArgumentException e) {
       System.out.println("Rule is not ready (probably being deleted).");
+      return true;
     }
+    return false;
   }
 
   public static String getInstanceStatus(String instanceName) throws IOException {

--- a/compute/cloud-client/src/test/java/compute/SnippetsIT.java
+++ b/compute/cloud-client/src/test/java/compute/SnippetsIT.java
@@ -291,8 +291,8 @@ public class SnippetsIT {
 
   public static void testListFirewallRules()
       throws IOException, ExecutionException, InterruptedException {
+    final PrintStream out = System.out;
     ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
-    PrintStream out = System.out;
     System.setOut(new PrintStream(stdOut));
     if (!isFirewallRuleDeletedByGceEnforcer(PROJECT_ID, FIREWALL_RULE_CREATE)) {
       compute.ListFirewallRules.listFirewallRules(PROJECT_ID);
@@ -306,8 +306,8 @@ public class SnippetsIT {
 
   public static void testPatchFirewallRule()
       throws IOException, InterruptedException, ExecutionException {
+    final PrintStream out = System.out;
     ByteArrayOutputStream stdOut = new ByteArrayOutputStream();
-    PrintStream out = System.out;
     System.setOut(new PrintStream(stdOut));
     if (!isFirewallRuleDeletedByGceEnforcer(PROJECT_ID, FIREWALL_RULE_CREATE)) {
       try (FirewallsClient client = FirewallsClient.create()) {

